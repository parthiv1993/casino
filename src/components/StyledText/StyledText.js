import PropTypes from 'prop-types';
import { Description, SubTitle, Title } from './StyledText.styles';

const StyledText = ({ children, type, ...restProps }) => {
  switch (type) {
    case 'title':
      return <Title {...restProps}>{children}</Title>;
    case 'subTitle':
      return <SubTitle {...restProps}>{children}</SubTitle>;
    case 'description':
      return <Description {...restProps}>{children}</Description>;
    default:
      return <Description {...restProps}>{children}</Description>;
  }
};

StyledText.propTypes = {
  children: PropTypes.node,
  type: PropTypes.string
};

export default StyledText;
