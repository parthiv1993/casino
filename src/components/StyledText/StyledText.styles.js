import styled from 'styled-components';

export const Title = styled.h2`
  font-weight: bold;
  font-size: 1.5rem;
  margin: 0;
  display: inline-block;
`;

export const SubTitle = styled.h4`
  font-weight: 600;
  font-size: 1.25rem;
  margin: 0;
  display: inline-block;
`;

export const Description = styled.p`
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
  display: inline-block;
`;
