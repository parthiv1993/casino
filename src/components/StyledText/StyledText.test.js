import { render } from '@testing-library/react';
import StyledText from './StyledText';

describe('StyledText', () => {
  it('should render title', () => {
    const { container } = render(<StyledText type="title">Dummy</StyledText>);
    expect(container).toMatchSnapshot();
  });

  it('should render subTitle', () => {
    const { container } = render(<StyledText type="subTitle">Dummy</StyledText>);
    expect(container).toMatchSnapshot();
  });

  it('should render description', () => {
    const { container } = render(<StyledText type="description">Dummy</StyledText>);
    expect(container).toMatchSnapshot();
  });

  it('should render description by default', () => {
    const { container } = render(<StyledText>Dummy</StyledText>);
    expect(container).toMatchSnapshot();
  });
});
