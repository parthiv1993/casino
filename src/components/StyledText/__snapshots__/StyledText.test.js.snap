// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText should render description 1`] = `
.c0 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
  display: inline-block;
}

<div>
  <p
    class="c0"
  >
    Dummy
  </p>
</div>
`;

exports[`StyledText should render description by default 1`] = `
.c0 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
  display: inline-block;
}

<div>
  <p
    class="c0"
  >
    Dummy
  </p>
</div>
`;

exports[`StyledText should render subTitle 1`] = `
.c0 {
  font-weight: 600;
  font-size: 1.25rem;
  margin: 0;
  display: inline-block;
}

<div>
  <h4
    class="c0"
  >
    Dummy
  </h4>
</div>
`;

exports[`StyledText should render title 1`] = `
.c0 {
  font-weight: bold;
  font-size: 1.5rem;
  margin: 0;
  display: inline-block;
}

<div>
  <h2
    class="c0"
  >
    Dummy
  </h2>
</div>
`;
